### ISSUE TITLE
dev mode/InjectManifest: registerSW.js not found / no precaching inserted 

### ISSUE THREAD

**Issue by urapadmin** [antfu/vite-plugin-pwa#368]
### Describe the bug

Not sure if this is a bug or rather me not understanding something. I am trying to use the plugin in dev mode with the injectManifest strategy. 
The plugin generates a registerSW.js in a sub folder called dev-dist. But the built in vite dev server does not serve from that directory. So I get a 404. 
When I manually move the file to the public folder the file gets served BUT it registers a dev-sw.js service worker that looks like my own sw.js: No precaching got injected. 

Is there a setting I overlook or is the dev mode not intended to test precaching? When building and using vite preview everything works fine.

My vite.config looks like this: 

``` javascript
import { defineConfig } from 'vite'
import {VitePWA } from 'vite-plugin-pwa'

// https://vitejs.dev/config/
export default defineConfig({
  build: {
  },
  plugins: [VitePWA({
    strategies: 'injectManifest',
    // srcDir: 'src',
    injectManifest: {
      globPatterns: ['**/*.html', '**/*.js', '**/*.css']
    },
    filename: 'sw.js',
    registerType: 'autoUpdate',
    devOptions: {
      enabled: true,
      type: 'module'
    }
  })]

})

```

and my sw.js is as simple as it gets (and that is the dev-sw.js I get when running vite in dev mode):
```
import { precacheAndRoute } from 'workbox-precaching'

precacheAndRoute(self.__WB_MANIFEST)
```

### Reproduction

(Happy to provide a repo if you think it is necessary in this case)

### System Info

```Shell
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz
    Memory: 4.18 GB / 15.78 GB
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.15.1 - ~\AppData\Roaming\npm\npm.CMD
  Browsers:
    Chrome: 104.0.5112.102
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by userquin** [comment]
@urapadmin pwa plugin version?

**Comment by urapadmin** [comment]
sorry, forgot to add the most important versions: pwa plugin is at 0.12.6, vite is at 3.0.7. 

Thanks for pointing me to the documentation but as far as I understand it I don't really need to add anything as long as I leave registering the sw / dev-sw.js entirely to the plugin, which I do. The generated code in the registerSW.js looks right to me.
 
As far as I can see the issue is really only that the request for the registerSW.js from somewhere within registerDevSW() does not succeed because the registerSW.js is generated in a sub-directory "dev-dist". If it were placed in the public directory or a location I could configure it would work just fine. Totally possible, that I am still not getting some basic concept here.

**Comment by userquin** [comment]
@urapadmin provide minimal repro

**Comment by urapadmin** [comment]
with pleasure: [https://github.com/arch-kiosk/simplelit](https://github.com/arch-kiosk/simplelit)
And many thanks for looking into it, I really hope I am not wasting your time by simply missing something obvious.

**Comment by userquin** [comment]
@urapadmin we need to modify the dev plugin...

**Comment by userquin** [comment]
check this repo (forked from your repo using #369):

![imagen](https://user-images.githubusercontent.com/6311119/188584895-87eff63d-4cf9-425a-a16d-1456d292f92c.png)

**Comment by urapadmin** [comment]
@userquin; Thank you, that was really fast. 
It works exactly as you show when I start it the first time. But only once. When I restart vite dev (having erased all active service workers and Chrome's cache) I can't get it to work again:
![image](https://user-images.githubusercontent.com/38838314/188650687-8207093f-609a-42e3-8b9c-a39fa808422c.png)

Perhaps this is a hint: After the first start and download of the app I get this line from the vite dev server:
![image](https://user-images.githubusercontent.com/38838314/188652455-43955fe4-7e73-4c10-928b-e1caafcb6746.png)

Even after that, as long as the same dev server instance keeps running, I can unregister my service worker, clear the cache and reload and I get a new service worker. But if I stop vite dev and just restart it, the above 404 is all I get.

**Comment by userquin** [comment]
@urapadmin thx for the info, I'll try to fix later

**Comment by userquin** [comment]
@urapadmin can you add `base:'/'` in the Vite root cofig and test again?

**Comment by urapadmin** [comment]
just to make sure, this is what I tried: 
```javascript
defineConfig({
  server: {
    base: '/'
  },
```
Alas, it did not change anything.

**Comment by userquin** [comment]
On root, the base for Vite, check my repo config file

**Comment by urapadmin** [comment]
I see. I get exactly the same result, though. registerSW.js can't be found. Now I use
```
export default defineConfig({
  base: '/',
  plugins: [VitePWA({
// ...
```

**Comment by userquin** [comment]
@urapadmin I'm fixing it: since we don't remove the dev-dist folder, if the `registerSW.js` file is already there we don't register it internally in a map, and so when requested it is missing...

**Comment by userquin** [comment]
@urapadmin updated my repo adding also the skip part to the sw (on udpdate will keep in skip waiting state): the local tgz file also updated, just remove the pwa plugin (`pnpm remove -D vite-plugin-pwa`), override the tgz with the new and install it again `pnpm add -D ./vite-plugin-pwa-0.12.6.tgz`

**Comment by urapadmin** [comment]
Good job, that fixed it. Works fine now in dev and preview.
Just for my understanding: While precaching works perfectly fine in the preview, no precaching happens in dev mode.``precacheAndRoute(self.__WB_MANIFEST)`` is simply turned into ``precacheAndRoute([])`` in the sw.js that vite dev serves. 
That is the intended behavior, I suspect?

**Comment by userquin** [comment]
@urapadmin released 0.12.7 version fixing this issue

**Comment by eokwukwe** [comment]
I am getting the same error as specified in this issue. I have tried using version `0.12.7` and `0.12.8` of the plugin. But the error remains. When I build and preview, I don't get the error.

```js
VitePWA({
  base: "/",
  devOptions: {
    enabled: true,
    type: "module",
    navigateFallback: "index.html",
  },
...
}),
```

**Comment by DodoDye** [comment]
I fixed the error with this:
```js
VitePWA({
    outDir: "./dev-dist",
    ...
})
```

**Comment by userquin** [comment]
> I am getting the same error as specified in this issue. I have tried using version `0.12.7` and `0.12.8` of the plugin. But the error remains. When I build and preview, I don't get the error. I am using the default strategy `generateSW`
> 
> ```js
> VitePWA({
>   base: "/",
>   devOptions: {
>     enabled: true,
>     type: "module",
>     navigateFallback: "index.html",
>   },
> ...
> }),
> ```

Open a new issue with a repo please
